src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -106,62 +134,64 @@ export default class PhotoField extends React.Component {
     }
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    const newView = this.state.src && (prevState.src !== this.state.src || (prevState.done && !this.state.done));
-    const cropper = this.refs.cropper;
-    if (cropper && newView) {
-      this.setCropBox();
-    }
-  }
-
   onEdit = () => {
     this.setState({ done: false, warningMessage: null });
   }
 
   onDone = () => {
-    this.setState({ done: true, warningMessage: null });
+    const cropResult = this.refs.cropper.getCroppedCanvas().toDataURL();
+    this.setState({ cropResult, done: true, warningMessage: null });
   }
 
   onChange = (files) => {
     if (files && files[0]) {
       const file = files[0];
-      const reader = new FileReader();
-      reader.readAsDataURL(file);
-      reader.onload = () => {
-        const fileName = file.name;
-        isValidImage(reader.result, fileName)
-          .then((isValid) => {
-            if (isValid) {
-              this.setState({
-                src: reader.result,
-                done: false,
-                cropResult: null,
-                errorMessage: null
-              });
-            } else if (!isValidFileType(file.name)) {
-              this.setState({
-                src: null,
-                done: false,
-                cropResult: null,
-                errorMessage: 'Please choose a file from one of the accepted types.'
-              });
-            } else if (isValidFileType(file.name)) {
+      if (file.preview) {
+        // dropzone recommendation
+        window.URL.revokeObjectURL(file.preview);
+      }
+      if (!isValidFileType(file.name)) {
```

Comment:
This `onChange` organization is much clearer, in fact, should we go ahead and remove the type validation as both our upload button and react-dropzone perform this automatically?

Reply:
I think we still need to do it here. The upload button's validation is more of a suggestion and can be easily avoided.